src/pages/dashboard/[id].jsx

```diff
@@ -0,0 +1,43 @@
+import { useSession } from 'next-auth/react'
+import { useState, useEffect } from 'react'
+import Roleselect from '@/pages/roleselect'
+import supabase from '@/lib/supabaseClient'
+
+const Dashboard = ({ role }) => {
+  const { data: session, status } = useSession()
+
+  if (status === 'unauthenticated') {
+    return <div>Access denied. Please log in. Redirect to login page then.</div>
+  }
+
+  if (status === 'loading') {
+    return <div>Loading page...</div>
+  }
+
+  return (
+    <>
+      {role === 'applicant' && <div>applicant dashboard</div>}
+      {role === 'recruiter' && <div>recruiter dashboard</div>}
+    </>
+  )
+}
+
+export async function getServerSideProps(context) {
+  const session = await getSession(context)
+  if (!session?.user?.id) {
```

Comment:
**Suggestion/Optional:**
We might be able to use Next.js middleware to protect our routes, just mentioning this as a suggestion.

https://next-auth.js.org/configuration/nextjs#middleware